ext/Draw/CDraw3DLine.cpp

```diff
@@ -0,0 +1,75 @@
+#include "../ext/Draw/CDraw3DLine.h"
+#include "../ext/Draw/Draw3DLineShaders.h"
+
+using namespace irr;
+using namespace video;
+using namespace scene;
+using namespace ext;
+using namespace draw;
+
+CDraw3DLine* CDraw3DLine::create(IVideoDriver* _driver)
+{
+    return new CDraw3DLine(_driver);
+}
+
+CDraw3DLine::CDraw3DLine(IVideoDriver* _driver)
+:   m_driver(_driver),
+    m_desc(_driver->createGPUMeshDataFormatDesc()),
+    m_meshBuffer(new IGPUMeshBuffer())
+{
+    auto callBack = new Draw3DLineCallBack();
+    m_material.MaterialType = (E_MATERIAL_TYPE)
+        m_driver->getGPUProgrammingServices()->addHighLevelShaderMaterial(
+        Draw3DLineVertexShader,
+        nullptr,nullptr,nullptr,
+        Draw3DLineFragmentShader,
+        3,EMT_SOLID,
```

Comment:
if you're going to draw lines I'd pass 2 instead of 3... but this is only cosmetic since that parameter is only needed for tessellation shaders.

Reply:
changed it anyways